sacred/observers/sql_bases.py

```diff
@@ -167,11 +195,18 @@ def get_or_create(cls, ex_info, session):
             Source.get_or_create(s, md5sum, session, ex_info["base_dir"])
             for s, md5sum in ex_info["sources"]
         ]
+        repositories = set()
+        for r in ex_info["repositories"]:
+            repository = Repository.get_or_create(r["url"], r["commit"], r["dirty"], session)
+            session.add(repository)
```

Comment:
Why do the repositories have to be added to the session, whereas the other objects are just obtained via `.get_or_create` ?

Reply:
```ex_info["repositories"]``` can contain multiple identical items (I believe, it's one item per source and multiple sources can come from the same repository) so if ```.get_or_create``` is called for each duplicate item before the first one is added to the session, the "repository" table ends up with duplicate repositories.

My understanding is that ```ex_info["dependencies"]``` and ```ex_info["base_dir"]``` wouldn't have duplicate items in a single experiment so it's enough to check if the items were added during previous experiments.